lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2507,6 +2507,86 @@ class ConvertAtenSqueezeOp : public OpConversionPattern<AtenSqueezeOp> {
 };
 } // namespace
 
+namespace {
+class ConvertAtenSqueezeDimOp : public OpConversionPattern<AtenSqueezeDimOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenSqueezeDimOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op.getLoc();
+    Value input = adaptor.self();
+    auto inputType = input.getType().cast<RankedTensorType>();
+    int64_t inputRank = inputType.getRank();
+    TypeConverter *typeConverter = getTypeConverter();
+    auto resultType =
+        typeConverter->convertType(op.getType()).cast<RankedTensorType>();
+    int64_t resultRank = resultType.getRank();
+
+    if (inputRank == 0) {
+      return rewriter.notifyMatchFailure(
+          op, "zero input rank should have been handled by the folder");
+    }
+
+    int64_t dim;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+      return rewriter.notifyMatchFailure(op, "dim must be constant");
+    if (dim < 0)
+      dim += inputRank;
+    if (dim < 0 || dim >= inputRank)
+      return rewriter.notifyMatchFailure(op, "dim is statically invalid");
+
+    // TODO: Add support for size-1 dynamic dimensions.
+    if (inputType.isDynamicDim(dim)) {
+      Value one = rewriter.create<arith::ConstantOp>(
+          loc, rewriter.getIntegerAttr(rewriter.getIndexType(), 1));
+      Value dimSize = getDimOp(rewriter, loc, input, dim);
+      Value dimSizeNotOne = rewriter.create<arith::CmpIOp>(
+          loc, arith::CmpIPredicate::ne, dimSize, one);
+      rewriter.create<AssertOp>(
+          loc, dimSizeNotOne,
+          rewriter.getStringAttr("unimplemented: dim(th) dimension should "
+                                 "not be size 1 dynamic"));
+    }
+
+    // If the dim(th) dimension of operand tensor type is not unit,
+    // `aten.squeeze` will behave as an identity operation.
+    if (inputType.getDimSize(dim) != 1) {
+      rewriter.replaceOpWithNewOp<tensor::CastOp>(op, resultType, input);
+      return success();
+    }
+
+    // In case the operand tensor type is of unit rank and is statically shaped
+    // with unit dimension, it will be collapsed to a 0-D tensor.
+    if (resultRank == 0) {
+      SmallVector<ReassociationIndices> reassociation;
+      rewriter.replaceOpWithNewOp<linalg::TensorCollapseShapeOp>(
+          op, resultType, input, reassociation);
```

Comment:
You can directly pass `ArrayRef<ReassociationIndices>()` instead of initializing `reassociation`, this will possibly save some stack space that SmallVector consumes by default.